### PR TITLE
Add Architectural Decision Records to US Notify

### PR DESCRIPTION
**Pull request by ccostino**
This changeset introduces Architectural Decision Records (ADRs) to the US Notify project.  This initial commit includes the following:

- Updates to the main project `README.md` that mentions ADRs
- A new `adr/` folder under the `docs/` folder
- An ADR template to copy and follow for the future
- Our first ADR, ADR-0001 - Establishing ADRs
- A `README.md` specific to ADRs that has more information and instructions

A few more tweaks are likely to come as we discuss this before the changeset is ready for full review.

## TODO

- [x] Update PR based on feedback
- [x] Add a note about updated the Architecture Decision Log in the maintenance section
- [x] [Create a separate issue to add or configure a GitHub issue template or form](https://github.com/GSA/notifications-api/issues/288) based on the new ADR template

Address #282

### CODE REVIEW

File: docs/adrs/README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+First, we have an ADR template that folks can use to work off of.  The template
+exists as both a GitHub issue template and a standalone Markdown file that can
+be copied as needed if folks prefer to work locally first.
+
+By following the template, we ensure that our ADRs are consistent in language
+and structure.  This allows us to easily review the documentions and discuss
+them as a team.  It also guarantees that the ADR has all of the required
+information.
+
+**ADRs are intended to be living documents.**  As such, it is not uncommon t
```
Comment (stvnrlly) [inline docs/adrs/README.md:50]:
```suggestion
**ADRs are intended to be living documents.**  As such, it is not uncommon to
```

**Review by stvnrlly — CHANGES_REQUESTED** [review]
Looks great so far! One thing that needs more info is what happens when we discuss/review an ADR. Could we add a section to the README between "Creating an ADR" and "Maintaining and ADR" about that?

File: docs/adrs/README.md (hunk 1 above)
Comment (ccostino) [reply docs/adrs/README.md:50]: D'oh!  I thought I caught them all... :-)

Thank you!

**Comment by ccostino** [comment]
Thank you!  And yes, that's a great catch about what happens during review/discussion!

I'll write something up and add that as a new section - I'm going to try and incorporate elements of the Art of Alignment training a lot of folks have gone through. :-)

File: docs/adrs/README.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+In either scenario, check to see what the latest ADR filename is, because they
+always start with a number (e.g., `0001`).  Name your ADR with a number one
+after the last ADR written; if the latest ADR starts with `0021-`, your ADR
+should start with `0022-`.
+
+At this point, it is a matter of filling in the details outlined in the template
+that are relevant to the ADR.
+
+
+### Reviewing an ADR
```
Comment (ccostino) [inline docs/adrs/README.md:79]: @stvnrlly here's the new section!

**Comment by ccostino** [comment]
I made a note above about updating the maintenance section to include a note about updated the Architecture Decision Log.

**Review by stvnrlly — APPROVED** [review]
🚀

**Pull request merged**